### DISCUSSION TITLE
How is cursor.sh able to show viewzones anywhere it wants

### DISCUSSION THREAD

**Discussion by Himanshu-Singh-Chauhan** [microsoft/vscode-discussions#1482] [Extension Development QnA] [answered]
like it shows history tab
![image](https://github.com/user-attachments/assets/d4904201-a17f-4c92-b2c2-a520e510cfe7)

it can show input box which is tiptapeditor a react component in terminal
![image](https://github.com/user-attachments/assets/16bafe01-8025-4b3e-b3f1-7f0ee231785e)

really curious, how can i do that in vscode source code, how to create view zones anywhere i like

**Comment by progrium** [comment, marked as answer]
not sure you can in vscode using built-in apis. interesting UI pattern though, where else have you seen it?

**Reply by Himanshu-Singh-Chauhan** [reply]
no where else, just cursor, but really interesting how they are modifying the vscode editor
